Added instructions / helpers to run on Canton

daravep opened the pull request.

daravep committed:
Added instructions / helpers to run on Canton

daravep committed:
Added script to conveniently run the chat app

daravep committed:
Fail script if solution was not built before

daravep committed:
Updated canton run script and dependencies

daravep committed:
Added lock file for startup

daravep committed:
Merge branch 'run_on_canton' of github.com:davidpadbury-da/daml-chat-app into run_on_canton

daravep committed:
Fixed lockfile behaviour

davidpadbury-da commented on src/App.js:
```diff
@@ -130,9 +130,9 @@ class App extends Component {
 
     const match = /\/(\w+) (.*)/.exec(newMessage)
-    const command = match ? match[1] : 'send'
+    const command = match ? match[1] : (newMessage === '/parties' ? 'parties' : 'send')
```
ah - so this is just doing `/parties` alone and in which case the regex isn't matching? perhaps update the expression to `\/(\w+)((?:\s*)(.*))?` and change content to `match[3]`.

daravep replied:
fixed

davidpadbury-da commented on src/App.js:
```diff
@@ -158,6 +158,14 @@ class App extends Component {
+      case 'parties':
+        const message = this.state.parties.map(x => x.partyId).join("\n")
+        this.setState({
+          messages: this.state.messages.concat({timestamp: new Date(), sender: 'Known Parties', text: message })
```
I think this message will be blown away the next time a message is sent

![shrug](https://media.giphy.com/media/jPAdK8Nfzzwt2/giphy.gif)

daravep replied:
Yeah, I know. But that was the only way I could find quickly without substantial changes.

davidpadbury-da reviewed (approved):
nice!

davidpadbury-da commented on README.md:
```diff
@@ -22,6 +22,25 @@ First run `npm install` to install required dependencies.
+
+First, run `npm install` to install the dependencies. 
+
+Then, run `npm build` to build the dar.
```
`npm run build`

daravep replied:
fixed

daravep committed:
Upgraded canton support to 0.3.0

Chat app failed to start with Canton 0.3.0 because `--bootstrap-script`
cli argument was renamed to `--bootstrap`. Fixed the run-on-canton.sh
script accordingly.

daravep committed:
Addressed reviewer comments
